message_sender/models.py

```
@@ -76,3 +77,28 @@ class OutboundSendFailure(models.Model):
 
     def __str__(self):  # __unicode__ on Python 2
         return str(self.id)
+
+
+@python_2_unicode_compatible
+class Channel(models.Model):
+
+    VUMI_TYPE = 'vumi'
+    JUNEBUG_TYPE = 'junebug'
+
+    CHANNEL_TYPES = (
+        (JUNEBUG_TYPE, 'Junebug'),
+        (VUMI_TYPE, 'Vumi')
+    )
+
+    channel_id = models.CharField(unique=True, null=False, blank=False,
+                                  max_length=64)
+    channel_type = models.CharField(choices=CHANNEL_TYPES, max_length=20,
+                                    default=JUNEBUG_TYPE)
+    concurrency_limit = models.IntegerField(null=False, blank=False, default=0)
+    message_delay = models.IntegerField(null=False, blank=False, default=0)
+    message_timeout = models.IntegerField(null=False, blank=False, default=0)
+    default = models.BooleanField(default=False)
```

We probably need to override the `save` method of this model, to make sure that if we are creating a new default model, that it sets `default=False` on the old default model.

Otherwise we'll have issues with 2 default models, and not knowing which one to use.

We probably also want to raise an error when setting a `default=False` on a default model, so that we can't end up in the state where we have no default models.

---

That combination would make it impossible to change the default channel though wouldn't it?

---

The idea would be when you save a channel with `default=True`, it would make all other channels `default=False`, so that there would only be one default channel.

That way, if you want to change the default channel, make another channel defaut, and the old default channel will be marked as not default.